src/core/Expander/ExpanderGroup.tsx

```diff
@@ -50,25 +54,116 @@ const OpenAllButton = ({
   return children;
 };
 
+const ExpanderGroupItems = (
+  children: Array<React.ReactElement<CompExpanderProps>>,
+) =>
+  React.Children.map(
+    children,
+    (child: React.ReactElement<CompExpanderProps>, index) => {
+      if (React.isValidElement(child)) {
+        const isChildOpen = child.props.open;
+        return React.cloneElement(child, {
+          index,
+          expanderGroup: true,
+          open: isChildOpen,
+        });
+      }
+      return child;
+    },
+  );
+
+const defaultProviderValue: CompExpanderProviderState = {
+  onExpanderOpenChange: () => null,
+  toggleAllExpanderState: {
+    toState: false,
+  },
+};
+const { Provider, Consumer: ExpanderGroupConsumer } = React.createContext(
+  defaultProviderValue,
+);
+
+const InitialStateOfExpanders = (
+  children: Array<React.ReactElement<CompExpanderProps>>,
+) => {
+  const openExpanders: number[] = [];
+  React.Children.forEach(children, (child, index) => {
+    if (React.isValidElement(child)) {
+      if (child.props.defaultOpen || child.props.open) {
+        openExpanders.push(index);
+      }
+    }
+  });
+  return openExpanders;
+};
+
 /**
  * <i class="semantics" />
  * Used for grouping expanders
  */
 export class ExpanderGroup extends React.Component<ExpanderGroupProps> {
-  render() {
-    const { OpenAll, CloseAll, ...passProps } = withSuomifiDefaultProps(
-      this.props,
+  state: ExpanderGroupState = {
+    openExpanders: InitialStateOfExpanders(this.props.children),
+    toggleAllExpanderState: {
+      toState:
+        InitialStateOfExpanders(this.props.children).length ===
+          this.props.children.length && this.props.children.length > 0,
+    },
+  };
+
+  handleExpanderOpenChange = (index: number = 0) => {
```

Comment:
Is the default parameter value really needed here? Should only apply changes when the index is really provided?

Reply:
Default parameter is now removed as it was not really needed, maybe never had been.